docker/run-docker

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+
+####################################################################################
+#
+#   Run this file from the root of the repository to build and run a PWA container
+#
+####################################################################################
+
+ENVFILE=./.env
+HOSTSFILE=/etc/hosts
+HOSTPRESENT=false
+
+main () {
+  # copy environment variables to .env in root for docker-compose to consume for build
+  message "Adding environment variables in root .env for docker setup."
+  cp ./docker/.env.docker $ENVFILE 
+  . $ENVFILE
+
+  DOMAIN=$PWA_STUDIO_PUBLIC_PATH
+  HOSTMAP="127.0.0.1 $DOMAIN"
+
+  add_domain_to_hostfile
+
+  generate_certificate
+
+  start_docker
+}
+
+add_domain_to_hostfile () {
```

Comment:
The `add_domain_to_hostfile` and `generate_certificate` functions look like they work great, but they seem to be duplicate work that [devcert](https://github.com/davewasmer/devcert) already does when we [call it from the Buildpack.Utilities.configureHost() function](https://github.com/magento-research/pwa-studio/blob/release/2.0/packages/pwa-buildpack/src/Utilities/configureHost.js#L75). It both updates the hostfile (cross-platform) and generates a CA and self-signed certificate. So could we just run a Node script here?

docker/makeHostAndCert.js
```js
const { Utilities: { configureHost } } = require('../packages/pwa-buildpack/');
const { writeFileSync } = require('fs');
const { join } = require('path');

const exactDomain = process.env.PWA_STUDIO_PUBLIC_PATH;

configureHost({
  exactDomain,
  addUniqueHash: false
})
.then(({ ssl: { key, cert }}) => {
  console.log('Hostfile updated and TLS certificate created and trusted. Saving...');
  save(key, 'key');
  save(cert, 'crt');
})
.catch(e => {
  console.error(e);
  process.exit(1);
});

function save(contents, ext) {
  const filePath = join('.', 'certs', `${exactDomain}.${ext}`);
  writeFileSync(filePath, 'utf8');
  console.log(`Saved ${filePath}`);
}
```

Then you could remove a lot of this code and just call `node makeHostAndCert.js` in this script.

When I run `run-docker`, I'm always on a machine with NodeJS, right? This is no knock on the shell script implementation itself, which looks fine, but the Node stuff is cross-platform and already tested.

Reply:
I see what you are saying here. The node script would depend on the ability of the user to be able to successfully run an `npm install` locally in order to use the library successfully to make the certs and host. I don't think we have that ability yet so I was keeping everything in a shell script outside of requiring any repo dependencies. With docker, you wouldn't need to require `node` to be on any user's machine, you could work out of docker exclusively and the only requirement would be having docker on your machine and the ability to clone a repository.

Reply:
I like the idea of this being immediately cross platform using devcert with your node script version of the above, but I ran into this error where one of the dependencies is saying that it is not valid on Win32: 

```
Error: \\?\C:\Users\User\code\pwa-studio\node_modules\node-zopfli-es\lib\binding\zop fli.node is not a valid Win32 application.                                           \\?\C:\Users\User\code\pwa-studio\node_modules\node-zopfli-es\lib\binding\zopfli.nod
```

And on my mac I ran into an error where it couldn't find the module by the path `../packages/pwa-buildpack`, not sure why. Let me know if there was anything else you did to get this to run for you locally. 

If this can solve the cross platform issue for docker then I'm for it but I think there is still more work for that next piece. I like that a bash script is not dependent on any installed node dependencies since docker builds are supposed to help with setting up a repeatable environment, the current state of it does not support Windows yet though so there's that.

Reply:
So the windows issue should be resolved if you rebase with master now that #864 is merged. Then you just need to `yarn install --ignore-optional`.

As for the mac issues I'm not sure, but I did find it necessary to run `yarn install`, then `yarn build` and _then_ any other commands like `run watch` because if you don't build it doesn't make thelinks.

Reply:
@sirugh did you try the node script specifically? I am still getting errors on Windows when I just try to run the node script. I've added support for Windows in the bash script for now. It made it a little busier than I would like but it covers getting a working installation up without a lot of fuss on the user's side. Windows does not make it easy to automate editing the host file or trusting certs.

Reply:
So after more thought, I don't think that it is a good idea to require an npm install in order to run a node script to generate a certificate and edit the host file. It makes the Docker setup have more dependencies when the only dependency _should_ be docker. Because we need https it requires having a trusted cert for a domain and I think that should be the only requirement added to setup docker, otherwise requiring an install before the docker install ends up duplicating work.

Reply:
> did you try the node script specifically?

I don't know, which one?

Reply:
I was referring to the one that James posted above.